README fixes/enhancements (#832)
NVIDIA/k8s-device-plugin: README.md

## Table of Contents

> - [NVIDIA device plugin for Kubernetes](#nvidia-device-plugin-for-kubernetes)

Reviewer: Question: What is the intent of this top-level element? It doesn't really seem to add value from a text / navigation perspective.

Author: It was some auto-format work that snuck in inadvertently. Removed.

compute resources to be explicitly partitioned and enforces these limits per
workload.

> With either time-slicing or MPS, the same sharing method is applied to all GPUs on

Reviewer: Nit:
```suggestion
With both time-slicing and MPS the same sharing method is applied to all GPUs on
```

#####  Single Config File Example
As an example, create a valid config file on your local filesystem, such as the following:

> ```yaml

Reviewer: why is this not `shell` when the example on line 696 is?

Author: Standardized.

For multiple config files, the procedure is similar.

Create a second `config` file with the following contents:

> ```yaml

Reviewer: Only part of this block is yaml. Strictly speaking this is `shell`.

Author: Yes, but the majority of the contents are YAML and that syntax highlights the contents the best, but happy to change if desired.

Reviewer 2: Hey @chipzoller , thanks for the PR,  Please change it to yaml

Author: It is currently set to yaml. Did you mean shell?

Reviewer 2: Ups yeah shell

Author: Done

`ConfigMap`), or used to override these options as desired.

These values are as follows:

> ```yaml

Reviewer: This isn't strictly yaml. Could you provide an example of how this renders?

Author: See example [here](https://github.com/chipzoller/nvidia-k8s-device-plugin/tree/cz-readme-fixes?tab=readme-ov-file#setting-other-helm-chart-values).

The most basic installation command without any options is then:

> ```sh

Reviewer: we use `shell` elsewhere in this document, could we use that here instead?

Author: Sure
